src/codegen/mod.rs

```diff
@@ -2095,10 +2117,27 @@ impl ToRustTy for FunctionSig {
             //     the array type derivation.
             //
             // [1]: http://c0x.coding-guidelines.com/6.7.5.3.html
-            let arg_ty = if let TypeKind::Array(t, _) = *arg_ty.canonical_type(ctx).kind() {
+
+
+            let arg_ty = match *arg_ty.canonical_type(ctx).kind()
```

Comment:
nit: indentation and spacing is quite odd in all this block right now, not sure if intended :)